src/clusto/services/http.py

```diff
@@ -201,12 +253,55 @@ def show(self, request, prefetch_attrs=None):
         for x in self.obj.attrs():
             attrs.append(unclusto(x))
         result['attrs'] = attrs
-        result['contents'] = [unclusto(x, prefetch_attrs) for x in self.obj.contents()]
-        result['parents'] = [unclusto(x, prefetch_attrs) for x in self.obj.parents()]
+        contents = self.obj.contents()
+        parents = self.obj.parents()
+
+        result['contents'] = mass_unclusto(contents, prefetch_attrs)
+        result['parents'] = mass_unclusto(parents, prefetch_attrs)
         result['actions'] = [x for x in dir(self) if not x.startswith('_') and callable(getattr(self, x))]
 
         return dumps(request, result)
 
+    def rename(self, request):
+        '''
+        Renames an object and returns the new object.
+
+        Example: /server/exampleserver/rename?object=/server/newname
+        '''
+        device_type, device_name = request.params['object'].strip('/').split('/')
+        if self.obj.type != device_type:
+            return Response(status=400, body='400 Renamed object must match type of original object\n')
+        if self.obj.name == device_name:
+            # The renamed object is the same as the current object, just show the object
+            return self.show(request)
+        try:
+            clusto.begin_transaction()
+            obj = clusto.get_by_name(device_name)
+            return Response(status=400, body='400 An entity with the new name "%s" already exists: %s\n' % (device_name, repr(obj)))
+        except LookupError, e:
```

Comment:
nit: it's a little bit weird to me that we are handling "correct" case with exception